### DISCUSSION TITLE
Should API ref template ref method/property structure be same as Javascript

### DISCUSSION THREAD

**Discussion by hamishwillee** [mdn/mdn-community#245] [Content] [answered]
The [API ref page template](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_reference_page_template) does not separate static and instance properties and methods - so it looks like this:

```
## Constructor
## Properties
## Methods
## Events
```

The Javascript docs have this structure (no documented template):

```
## Constructor
## Static properties
## Static methods
## Instance properties
## Instance methods
```

Should the API ref template match Javascript in this respect? It seems to me that the distinction is useful, so we should match them.

This comes out of https://github.com/mdn/content/pull/21241

**Comment by meyerweb** [comment]
For what it’s worth, I sort of followed the Javascript pattern in documenting the Temporal API (e.g. https://github.com/meyerweb/content/blob/temporal/files/en-us/web/javascript/reference/global_objects/temporal/calendar/index.md), and it made sense to me.  So I’m in favor.

**Reply by hamishwillee** [reply]
Thanks! FYI, irrespective of whether we agree the structure, the headings should lower case the second word - so "Instance properties" rather than "Instance Properties".

**Reply by meyerweb** [reply]
Ah, whoops; thanks!  Hopefully Temporal will advance a bit more and I can make those corrections to get the docs actually merged!

**Comment by wbamberg** [comment]
I think the Web/API interface pages should split static and instance methods/properties, yes.

We can use page types to know which are which :) and also https://docs.google.com/spreadsheets/d/14RX8EEKPpeYkP2cF5Y_ZKF3JpNJ_24OL7stvZWNJLIY/edit#gid=2070323159.

**Reply by hamishwillee** [reply]
OK. So I've created https://github.com/mdn/content/pull/21273 to update the page template. If that gets accepted I'll kick off migrating everything. I don't think it likely anyone will object strongly.

@schalkneethling How do I kick of getting the linter to match the agreed page type structure once everyone is behind this? (and initial migration complete).

**Reply by hamishwillee** [reply]
And who else do we need to loop into the discussion? @teoli2003 ?

**Reply by teoli2003** [reply]
This looks fine to me. 👍

**Comment by hamishwillee** [comment, marked as answer]
The answer is that this is agreed - Web API interface pages will split up the types to static/instance for properties and methods. Page template updated (might not render yet): https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_reference_page_template

I'll kick off a mini project to fix up the whole web API as it is now - later in the week.
